engine/minskyTensorOps.cc

```diff
@@ -162,6 +162,10 @@ namespace minsky
 //  };
 
   template <OperationType::Type op> struct GeneralTensorOp;
+
+
+  template <OperationType::Type ravel> struct RavelTensor;  
```

Comment:
Why does this need to be a template?

Reply:
This is still the part of the first commit. I'll commit the latest code with ravel as a registered single op in tensorOpFactory, but which still needs the actual operations. I'm going to spend the morning finishing working through Ravel's source code to get a complete picture of how the operations are invoked (with sorting and other ancillary operations performed on the data before the actual chain of operations in RawData::reduce()). I envision that I will start adding the operations in minskyTensorOps.cc this evening.

Reply:
The latest version also does not access the actual ravel state. I only managed to achieve that once today: by adding a RavelTensorOp struct to minskyTensorOps.cc, one deriving from EvalOp<OperationType::ravel>, to a sligthly modified version of yours and just before the RavelTensor class. In that case, it worked like the previous dense tensor version only for simple tensors such as iota(9). It would print out the value as the caliper was moved along the single handle and the sum of 36 (for 0, 1, 2, 3, 4, 5, 6, 7, 8) when the handle was collapsed. It would of course not work for a higher dimensional case (the dimensions would be all wrong). My idea was to have the dense tensor version working like before at least. But I now understand why we can't have this version fall back to 2.15.5 for even dense tensors...